### ISSUE TITLE
Unfortunately, quite slow to update the bridges for the Google Play Services components

### ISSUE THREAD

**Issue by horaciojcfilho** [xamarin/GooglePlayServicesComponents#251]
Please I love the Xamarin.Android product, but the updates to the bridges for the Google Play Services components happen quite slow.

For example:
- Firebase Messaging
-- Current version of Xamarin.Android bridge: 17.4.0 (Feb 23, 2019)
-- Latest version of Java library: 19.0.1 (Jun, 2019)
- Play Services Location
-- Current version of Xamarin.Android bridge: 16.0.0 (Oct, 2018)
-- Latest version of Java library: 17.0.0 (Jun, 2019)
- Play Services Maps
-- Current version of Xamarin.Android bridge: 16.0.0 (Oct, 2018)
-- Latest version of Java library: 17.0.0 (Jun, 2019)

These are some examples of how the current version of the bridges available on Nuget can be far from the latest version of their Java counterparts. Please sorry for any inconvenience around.

**Comment by horaciojcfilho** [comment]
Hello @Redth and @moljac,

Please don't take me wrong, but it is very hard to rely on Xamarin.Android with such dependencies (AndroidX and Google Play Services) taking so long to get updated and being constantly out of date :(

I apologize for any inconvenience.

**Comment by moljac** [comment]
@horaciojcfilho 
No offense taken, but it would help us a lot if you could specify exactly what is your problem.

AndroidX is in preview check 

https://www.nuget.org/packages?q=Xamarin.AndroidX.* 

Note nuget search will return not only AndroidX packages, but you will find them for sure.

Preview and Release for bindings is almost the same. Release has been out as preview for a while so we fixed some main issues (missing classes and methods and maybe some runtime issues).

We are right now maintaining Android.Support, preparing (updating) AndroidX for release, maintaining GPS-FB with Android.Support dependencies and binding new GPS-FB with AndroidX dependencies. 

Bottom line in last 2 years our work doubled just on Android.Support and GPS-FB side.

Can you specify what is "out of date"?

Thanks

**Comment by horaciojcfilho** [comment]
Hello @moljac,

Thank you very much for getting in touch. By "out of date" I would like to mean not being the latest version available, please sorry if I have been exaggerated in this point.

For example, the preview Xamarin.AndroidX bindings are no longer matching the latest version available:
- AndroidX AppCompat
-- Current version of Xamarin.Android binding: 1.0.2 (stable version released in Nov, 2018)
-- Latest version of Java library: 1.1.0 (alpha version started in Dec, 2018, stable version released in Sep, 2019)
- AndroidX Core
-- Current version of Xamarin.Android binding: 1.0.1 (stable version released in Nov, 2018)
-- Latest version of Java library: 1.1.0 (alpha version started in Nov, 2018, stable version released in Sep, 2019)

Please could the mantainance of the bindings start with the alpha version of the Java libraries? The idea is to get the bindings almost ready near to the stable version release, which is likely to reduce the delay to get Xamarin.Android and Java libraries matching in terms of the latest version available. 

Sorry again for any inconvenience.

**Comment by Redth** [comment]
@horaciojcfilho we're working as quickly as we can to update Google Play Services / Firebase and AndroidX.  As we are nearing stable release of our AndroidX bindings, we can transition to newer versions of Google Play Services / Firebase which have native AndroidX dependencies.  Over the past year we've made a lot of changes to our processes and tooling which I'm confident will allow us to speed up the release cadence of new versions.  I can appreciate you are one of the few who want them as fast as possible, and we are doing our best to prioritize resources accordingly.  

Thanks for your patience!

**Comment by horaciojcfilho** [comment]
Hello @moljac and @Redth,

Please I pretty appreciate such an exciting feedback. I apologize again if I has been somehow rude, that was not my idea. Xamarin.Android is a tool that has helped a lot to prototype Android-related stuff :)

Thank you for the great work and thank you for the feedback.

**Comment by moljac** [comment]
@horaciojcfilho 

Published 71.20190927 few minutes ago. I'm not sure if that solves your problems, but it will solve issues for some other users.

I'm working on further issues and GPS-FB with AndroidX dependencies.

Closing this issue.
Thanks
